cli/lib/args.rs

```diff
@@ -167,22 +168,60 @@ pub enum NpmProcessStateKind {
 
 pub static NPM_PROCESS_STATE: LazyLock<Option<NpmProcessState>> =
   LazyLock::new(|| {
+    /// Allows for passing either a file descriptor or file path.
+    enum FdOrPath {
+      Fd(usize),
+      Path(PathBuf),
+    }
```

Comment:
Changed this to allow reading the npm process state if it's a file path because it will allow JS/Wasm code to write the data to a temp file then pass the path (since it doesn't have access to the raw fd)